Content.Server/GameObjects/EntitySystems/Click/InteractionSystem.cs

```diff
@@ -192,222 +214,136 @@ private bool HandleWideAttack(ICommonSession? session, EntityCoordinates coords,
             {
-                Logger.WarningS("system.interaction",
-                    $"Client sent pull interaction with client-side entity. Session={session}, Uid={uid}");
-                return false;
+                Logger.InfoS("system.interaction", $"TryPullObject input validation failed");
+                return true;
             }
 
-            var player = session?.AttachedEntity;
-
-            if (player == null)
-            {
-                Logger.WarningS("system.interaction",
-                    $"Client sent pulling interaction with no attached entity. Session={session}, Uid={uid}");
+            if (userEntity.Uid == uid)
                 return false;
-            }
 
             if (!EntityManager.TryGetEntity(uid, out var pulledObject))
-            {
                 return false;
-            }
 
-            if (player == pulledObject)
-            {
+            if (!InRangeUnobstructed(userEntity, pulledObject, popup: true))
                 return false;
-            }
 
             if (!pulledObject.TryGetComponent(out PullableComponent? pull))
-            {
                 return false;
-            }
-
-            var dist = player.Transform.Coordinates.Position - pulledObject.Transform.Coordinates.Position;
-            if (dist.LengthSquared > InteractionRangeSquared)
-            {
-                return false;
-            }
 
-            return pull.TogglePull(player);
+            return pull.TogglePull(userEntity);
         }
 
-        public async void UserInteraction(IEntity player, EntityCoordinates coordinates, EntityUid clickedUid)
+        public async void UserInteraction(IEntity user, EntityCoordinates coordinates, EntityUid clickedUid)
         {
-            // Get entity clicked upon from UID if valid UID, if not assume no entity clicked upon and null
-            if (!EntityManager.TryGetEntity(clickedUid, out var attacked))
-            {
-                attacked = null;
-            }
-
-            // Verify player has a transform component
-            if (!player.TryGetComponent<ITransformComponent>(out var playerTransform))
+            if (user.TryGetComponent(out CombatModeComponent? combatMode) && combatMode.IsInCombatMode)
             {
+                DoAttack(user, coordinates, false, clickedUid);
                 return;
             }
 
-            // Verify player is on the same map as the entity he clicked on
-            if (coordinates.GetMapId(_entityManager) != playerTransform.MapID)
+            // Verify user is on the same map as the entity he clicked on
+            if (coordinates.GetMapId(_entityManager) != user.Transform.MapID)
             {
                 Logger.WarningS("system.interaction",
-                    $"Player named {player.Name} clicked on a map he isn't located on");
+                    $"User entity named {user.Name} clicked on a map he isn't located on");
                 return;
```

Comment:
Should this be included as part of ValidateClientInput?

Reply:
It could, but putting it here also checks input from the server. ValidateClientInput only checks Handlers from messages that come from the client.